### PR TITLE
[dbnode] Fix wrong metrics calculation for block iterator

### LINKED ISSUE #3623: [dbnode] wrong metrics calculation for block iterator
@ryanhall07 @robskillington Just verifying with you as I read this code multiple times, and I think I found a bug:
size is the number of elements in `results` as effectively returned originally here: 
```
func (r *results) AddDocuments(batch []doc.Document) (int, int, error) {
	r.Lock()
	err := r.addDocumentsBatchWithLock(batch)
	size := r.resultsMap.Len()
	docsCount := r.totalDocsCount + len(batch)
	r.totalDocsCount = docsCount
	r.Unlock()
	return size, docsCount, err
}
```

Since `results` only keeps growing, as I haven't seen any reset to this object, it means that when you call iter.Counts() to get the seriesMatched, you will get a wrong number, since the AddSeries(size) keeping adding up total and not adding delta. Maybe the correct way is to measure results before adding and after to get that delta?

_Originally posted by @asafm in https://github.com/m3db/m3/pull/3269#discussion_r669622217_

### PR DESCRIPTION
**Pull request by asafm**
**What this PR does / why we need it**:
AddSeries and AddDocs were expecting a delta integer as they do += inside, hence this fix is to pass the change in both series and docs due to the results being added docs in this method of queryWithSpan on block

Fixes #3623

### CODE REVIEW

**Comment by asafm** [comment]
@ryanhall07 Can you review this please?

**Comment by asafm** [comment]
@robskillington who can help with this PR?

File: src/dbnode/storage/index/block.go [hunk 1]
```diff
@@ -528,8 +530,8 @@ func (b *block) queryWithSpan(
 		}
 	}
 
-	iter.AddSeries(size)
-	iter.AddDocs(docsCount)
+	iter.AddSeries(size - sizeBefore)
```
Comment (ryanhall07) [inline src/dbnode/storage/index/block.go:533]: can we add a test to verify the change?

**Review by ryanhall07 — APPROVED** [review]

File: src/dbnode/storage/index/block.go (hunk 1 above)
Comment (asafm) [reply src/dbnode/storage/index/block.go:533]: I wasn't sure if I should since it didn't exists before. I will add.

**Comment by wesleyk** [comment]
@asafm can you update the branch and we'll get it merged?

**Comment by asafm** [comment]
@wesleyk Sorry for the delay - August vacations - I'm merged from master

**Comment by wesleyk** [comment]
> @wesleyk Sorry for the delay - August vacations - I'm merged from master

Thanks! Could we add a unit test to verify behavior as well?

File: src/dbnode/storage/index/block.go (hunk 1 above)
Comment (asafm) [reply src/dbnode/storage/index/block.go:585]: @ryanhall07 Added a unit test

**Comment by asafm** [comment]
@wesleyk Added a unit test

**Pull request merged**